### DISCUSSION TITLE
[firebase_dynamic_links] Navigate to dynamic link from within app

### DISCUSSION THREAD

**Discussion by ozzy1873** [firebase/flutterfire#5569] [Feature request]
I added dynamic link support to my app and it works great when used as intended. However, one of my users copied the dynamic link to the clipboard and pasted it into a chat section of the same app that generated it. When the dynamic link is tapped, an error page is displayed stating "Dynamic Link Not Found". Pasting the same link into a different app, like the Chrome browser, works as intended. I am using url_launcher 5.2.3 to navigate to the dynamic link.

It would be nice to be able to use dynamic links to navigate within the same app that generates them. If there was a method to convert a short dynamic link to the long form, I would be able to work around this issue. I suppose I could use the full url instead of short links, but it would be nice to keep the messages short.

**Comment by iapicca** [comment]
Hi @ozzy1873 
can you please provide your updated  `flutter doctor -v` ?
Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.
Thank you

**Comment by ozzy1873** [comment]
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version 10.0.18362.418], locale en-US)
    • Flutter version 1.9.1+hotfix.6 at c:\src\flutter
    • Framework revision 68587a0916 (6 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\john\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 40.2.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Pixel 3 • 8B8X15F76 • android-arm64 • Android 10 (API 29)

• No issues found!

**Comment by ozzy1873** [comment]
I'll see if I can come up with a simple app.
